native/cocos/platform/openharmony/OpenHarmonyPlatform.cpp

```diff
@@ -0,0 +1,379 @@
+                DECLARE_NAPI_FUNCTION("onReady", OpenHarmonyPlatform::NapiOnReady),
+                DECLARE_NAPI_FUNCTION("onShow", OpenHarmonyPlatform::NapiOnShow),
+                DECLARE_NAPI_FUNCTION("onHide", OpenHarmonyPlatform::NapiOnHide),
+                DECLARE_NAPI_FUNCTION("onDestroy", OpenHarmonyPlatform::NapiOnDestroy),
+                DECLARE_NAPI_FUNCTION("onActive", OpenHarmonyPlatform::NapiOnActive),
+                DECLARE_NAPI_FUNCTION("onInactive", OpenHarmonyPlatform::NapiOnInactive),
+            };
+            NAPI_CALL(env, napi_define_properties(env, exports, sizeof(desc) / sizeof(desc[0]), desc));
+#endif
+        } break;
+        case XCOMPONENT_REGISTER_LIFECYCLE_CALLBACK: {
+            LOGE("kee cocos GetContext XCOMPONENT_REGISTER_LIFECYCLE_CALLBACK");
+        } break;
+        case NATIVE_RENDER_API: {
+            LOGE("kee cocos GetContext NATIVE_RENDER_API");
+            napi_property_descriptor desc[] = {
+                DECLARE_NAPI_FUNCTION("nativeEngineInit", OpenHarmonyPlatform::NapiNativeEngineInit),
+            };
+            NAPI_CALL(env, napi_define_properties(env, exports, sizeof(desc) / sizeof(desc[0]), desc));
+        } break;
+        case WORKER_INIT: {
+            LOGE("kee cocos GetContext WORKER_INIT");
+            napi_property_descriptor desc[] = {
+                DECLARE_NAPI_FUNCTION("workerInit", OpenHarmonyPlatform::NapiWorkerInit),
+            };
+            NAPI_CALL(env, napi_define_properties(env, exports, sizeof(desc) / sizeof(desc[0]), desc));
+        } break;
+        case ENGINE_UTILS: {
+            LOGE("kee cocos GetContext ENGINE_UTILS");
+            napi_property_descriptor desc[] = {
+                DECLARE_NAPI_FUNCTION("resourceManagerInit", OpenHarmonyPlatform::NativeResourceManagerInit),
+            };
+            NAPI_CALL(env, napi_define_properties(env, exports, sizeof(desc) / sizeof(desc[0]), desc));
+        }
+        case UV_ASYNC_SEND: {
+            LOGE("kee cocos GetContext UV_ASYNC_SEND");
+            napi_property_descriptor desc[] = {
+                DECLARE_NAPI_FUNCTION("send", OpenHarmonyPlatform::NapiASend),
+            };
+            NAPI_CALL(env, napi_define_properties(env, exports, sizeof(desc) / sizeof(desc[0]), desc));
+        } break;
+        default:
+            LOGE("kee cocos unknown type");
+    }
+    return exports;
+}
+
+OpenHarmonyPlatform* OpenHarmonyPlatform::getInstance() {
+    return dynamic_cast<OpenHarmonyPlatform*>(BasePlatform::getPlatform());
+}
+
+OpenHarmonyPlatform::OpenHarmonyPlatform() {
+}
+
+int OpenHarmonyPlatform::getSdkVersion() const {
+    return 0;
+}
+
+int32_t OpenHarmonyPlatform::run(int argc, const char** argv) {
+    if (workerLoop_) {
+        uv_timer_init(workerLoop_, &timerHandle_);
+        // 1s = 1000ms = 60fps;
+        // 1000ms / 60fps = 16 ms/fps
+        uv_timer_start(&timerHandle_, &OpenHarmonyPlatform::TimerCb, 16, true);
```

Comment:
Repeat timer in libuv is not inaccurate and unreliable. We should setup timer for every frame, and adjust delay duration frame by frame.

Reply:
This issue can be fixed by using libuv interfaces, and it is not urgency. We can add todo here and fix it later